zebra-chain/src/orchard/note/nullifiers.rs

```diff
@@ -46,9 +49,19 @@ impl Hash for Nullifier {
     }
 }
 
```

Comment:
```suggestion
#[cfg(test)]
impl From<[u64; 8]> for Nullifier {    
    fn from(bytes: [u64; 8]) -> Self {        
        Self(pallas::Base::from_u512(bytes))
    }
}
        
```

Reply:
I like the idea however it seems `from_u512` is not public:

https://github.com/zcash/pasta_curves/blob/93df9c0cb9241e788916bca8e9a509b0b03f6023/src/fields/fq.rs#L254
https://github.com/zcash/pasta_curves/blob/93df9c0cb9241e788916bca8e9a509b0b03f6023/src/fields/fp.rs#L254

Do you think we should submit a PR to `pasta_curves` to make these public ?

Reply:
Use `FieldExt::from_bytes_wide` instead. (Trait methods are automatically public.)
https://github.com/zcash/pasta_curves/blob/93df9c0cb9241e788916bca8e9a509b0b03f6023/src/fields/fq.rs#L715

Rather than adding a `From` impl, I'd prefer to just use `FieldExt::from_bytes_wide` in the `Arbitrary` impl.

If we `impl From<[u64; 8]>`, then we could accidentally read 64 bytes during deserialization, and turn it into a `Nullifier` using `into`. That would be really hard to pick up in a code review. (We could also easily delete a `#[cfg(test)]`, particularly if it doesn't have any usage comments.)

Reply:
> And rather than adding a From impl, I'd prefer to just use FieldExt::from_bytes_wide in the Arbitrary impl.

Yeah I like this

Reply:
OK so, ignore this suggestion, and if we use `pallas::Base::from_bytes_wide()` inside the `Arbitrary` impl, we don't need a `TryFrom` impl for `orchard::Nullifier`, just the regular `From` for 32 bytes for now. Marking this convo thread resolved (but open to comments if I've missed anything)

Reply:
It looks to me that the `TryFrom` is still needed because want a parse error if something invalid gets deserialized https://github.com/ZcashFoundation/zebra/blob/854f32d1f9e133f928a8670675c304882b92b344/zebra-chain/src/orchard/action.rs#L64

Reply:
I think @oxarbitrage is right:
- parsing needs `TryFrom` because it deserializes potentially malicious 256-bit data from the network
- `Arbitrary` needs `from_bytes_wide` because it turns 512 bits of random data into a valid near-uniform random point

Reply:
Applied @teor2345's suggested change and I like how this looks now